src/core/file-impl.hh

```diff
@@ -276,8 +282,12 @@ public:
 };
 
 class blockdev_file_impl final : public posix_file_impl {
+    int ng_device_fd;
+    uint32_t nsid;
+    size_t block_size;
```

Comment:
It's worth grouping nsid and block_size on a `struct nvme_device_info` (or alike) and carry them around altogether

Reply:
Okay, we fixed it.